EventFilter/L1TRawToDigi/plugins/implementations_stage2/CaloSummaryUnpacker.cc

```diff
@@ -0,0 +1,71 @@
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "EventFilter/L1TRawToDigi/plugins/UnpackerFactory.h"
+
+#include "L1Trigger/L1TCalorimeter/interface/CaloTools.h"
+
+#include "L1TObjectCollections.h"
+
+#include "DataFormats/L1CaloTrigger/interface/CICADA.h"
+
+#include "CaloSummaryUnpacker.h"
+#include "GTSetup.h"
+
+#include <cmath>
+
+float l1t::stage2::CaloSummaryUnpacker::processBitsToScore(const unsigned int bitsArray[]) {
+  float constructedScore = 0.0;
+  //All bits have been shifted to just left of the decimal point
```

Comment:
Actually, reading the formula below I think the bits have been shifted so that
  - the bits go from `0x0000` to `0xffff`
  - the score goes from `0` to `255.996...`

Reply:
yes, the maximum score of CICADA is 0xff.ff ~= 255.996

Reply:
OK, then I think my simplified approach [above](https://github.com/cms-sw/cmssw/pull/44222#pullrequestreview-2139049212) should be correct.

Reply:
It just wouldn't surprise me if the unpacker formula was over complicated, I was just writing off the top of my head, but it also demonstrably works. If it is going to be a sticking point, I can rework it to be less complicated.

Reply:
Try the simplified version...

Reply:
The problem is not only that the code works, but also that somebody will have to maintain it over the next years.

Reply:
By the way, there seems to be some duplication between this code and that in https://github.com/cms-sw/cmssw/pull/45294 .

Maybe some can be reused ?

Reply:
The simplification has been implemented, but there is a broader discussion to be had now about what happens to https://github.com/cms-sw/cmssw/pull/45294